numpydoc/numpydoc.py

```diff
@@ -184,6 +183,31 @@ class NumpyCDomain(ManglingDomainBase, CDomain):
     }
 
 
+def match_items(lines, content_old):
+    """Create the right items for lines.
+
+    This tries to recover where a line in ``lines``
+    came from before mangling.
+
+    It assumes that missing or new lines
+    are always empty.
```

Comment:
I'm not sure the idea here is really correct, given that the mangling can do other stuff than only adding empty lines.
(Iirc e.g. plot directive insertion).